File: app/src/main/java/com/zf/spycamera/activity/SettingActivity.java
package com.zf.spycamera.activity;

import android.content.Intent;
import android.os.Bundle;
import android.provider.Settings;
import android.support.annotation.Nullable;
import android.support.v7.app.AppCompatActivity;
import android.util.Log;
import android.view.View;
import android.widget.ImageView;
import android.widget.RelativeLayout;

import com.zf.spycamera.FloatWindowService;
import com.zf.spycamera.R;
import com.zf.spycamera.utils.FloatWindowUtil;
import com.zf.spycamera.utils.SPUtil;
import com.zf.spycamera.view.PreviewView;

import java.util.Set;

/**
 * Created by Allever on 18/5/11.
 */

public class SettingActivity extends AppCompatActivity implements View.OnClickListener {
    private static final String TAG = "SettingActivity";
    private RelativeLayout mRlPreviewCont;
    private ImageView mIvPreview;

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_setting);

        initData();
        initView();
    }

    private void initData(){

    }

    private void initView(){
        mRlPreviewCont = findViewById(R.id.id_setting_rl_preview_container);
        mIvPreview = findViewById(R.id.id_setting_iv_prieview);
        mRlPreviewCont.setOnClickListener(this);
        mIvPreview.setOnClickListener(this);

        refreshCb();
    }

    private void refreshCb(){
        if (SPUtil.getShowPreview(this)){
            Log.d(TAG, "refreshCb: isShow = " + SPUtil.getShowPreview(this));
            mIvPreview.setImageResource(R.drawable.checkbox_on);
        }else {
            mIvPreview.setImageResource(R.drawable.checkbox_off);
        }
    }

    @Override
    public void onClick(View v) {
        int id = v.getId();
        switch (id){
            case R.id.id_main_ll_menu_container:
            case R.id.id_setting_iv_prieview:
                SPUtil.setShowPreview( SettingActivity.this,!SPUtil.getShowPreview(SettingActivity.this));
                refreshCb();
                if (FloatWindowService.mService != null){
                    //相机已启动
                    Intent intent = new Intent(this, FloatWindowService.class);
                    stopService(intent);
                    startService(intent);
                }
                break;
            default:
                break;
        }
    }
}

File: app/src/main/java/com/allever/stealthcamera/ui/SettingActivity.kt
package com.allever.stealthcamera.ui

import android.content.Intent
import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import android.view.View
import android.widget.ImageView
import com.allever.lib.common.util.FeedbackHelper

import com.allever.stealthcamera.FloatWindowService
import com.allever.stealthcamera.R
import com.allever.stealthcamera.utils.SPUtil

/**
 * Created by Allever on 18/5/11.
 */

class SettingActivity : AppCompatActivity(), View.OnClickListener {
    private lateinit var mIvPreview: ImageView
    private lateinit var mIvFrontCameraSwitch: ImageView

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_setting)

        initData()
        initView()
    }

    private fun initData() {

    }

    private fun initView() {
        findViewById<View>(R.id.id_setting_rl_preview_container).setOnClickListener(this)
        mIvPreview = findViewById(R.id.id_setting_iv_prieview)
        mIvPreview.setOnClickListener(this)

        findViewById<View>(R.id.id_setting_rl_camera_container).setOnClickListener(this)
        mIvFrontCameraSwitch = findViewById(R.id.id_setting_iv_front_camera)
        mIvFrontCameraSwitch.setOnClickListener(this)

        setSwitch(mIvPreview, SPUtil.getShowPreview(this))
        setSwitch(mIvFrontCameraSwitch, SPUtil.getUseFrontCamera(this))

        findViewById<View>(R.id.id_setting_rl_feedback_container).setOnClickListener(this)
        findViewById<View>(R.id.id_setting_rl_about_container).setOnClickListener(this)
    }

    private fun setSwitch(target: ImageView, switch: Boolean) {
        if (switch) {
            target.setImageResource(R.drawable.ic_preview_on)
        } else {
            target.setImageResource(R.drawable.ic_switch_off)
        }

    }

    override fun onClick(v: View) {
        when (v.id) {
            R.id.id_main_ll_menu_container, R.id.id_setting_iv_prieview -> {
                SPUtil.setShowPreview(this@SettingActivity, !SPUtil.getShowPreview(this@SettingActivity))
                setSwitch(mIvPreview, SPUtil.getShowPreview(this))
                restartFloatService()
            }

            R.id.id_setting_iv_front_camera -> {
                SPUtil.setUseFrontCamera(this@SettingActivity, !SPUtil.getUseFrontCamera(this@SettingActivity))
                setSwitch(mIvFrontCameraSwitch, SPUtil.getUseFrontCamera(this))
                restartFloatService()
            }
            R.id.id_setting_rl_feedback_container -> {
                FeedbackHelper.feedback(this)
            }
            R.id.id_setting_rl_about_container -> {
                AboutActivity.start(this)
            }
            else -> {
            }
        }
    }

    private fun restartFloatService() {
        if (FloatWindowService.mService != null) {
            //相机已启动
            val intent = Intent(this, FloatWindowService::class.java)
            stopService(intent)
            startService(intent)
        }
    }

    companion object {
        private const val TAG = "SettingActivity"
    }
}
